### DISCUSSION TITLE
How to add a RayCaster sensor in Isaac Sim?

### DISCUSSION THREAD

**Discussion by shaoxiang** [isaac-sim/IsaacLab#1577] [Q&A]
Thank you very much for Isaac Lab's excellent robot learning platform.
I trained a quadruped robot on rough terrain in Isaac Lab and now I want to migrate the policy to Isaac Sim, but I am having trouble adding sensor observations.
Isaac Sim doesn't seem to have a RayCaster sensor, only something like a LiDAR or ultrasonic sensor. 
How can I get a RayCaster ground height measurement sensor in Isaac Sim?

**Comment by RandomOakForest** [comment]
Thank you for posting this question. There is a [generic ray sensor in Isaac Sim](https://docs.omniverse.nvidia.com/isaacsim/latest/advanced_tutorials/tutorial_advanced_range_sensor_generic.html#isaac-sim-app-tutorial-advanced-range-sensor-generic) that can be configured to approximate a ray caster.  The complete list of available [sensors is found in this doc](https://docs.omniverse.nvidia.com/isaacsim/latest/features/sensors_simulation/index.html). Also, you may consider a LiDAR sensor.

**Reply by shaoxiang** [reply]
Thanks. I will try range_sensor in isaac sim. 
I can imagine that this work is difficult because the sensor data is a little bit different. So I think the sim2sim work from isaac lab to isaac sim is not simple, right?
I found the same problem. There is no set_external_force_and_torque method in Articulation in isaac sim.
How to sim2sim quadcopter policy is also a question. 
Hopefully there will be more examples to explain this...

**Reply by shaoxiang** [reply]
In Isaac Sim, you can use
**from isaacsim.core.prims import SingleArticulation**
to implement **Articulation** in Isaac Lab
RayCaster sensor seems to be implemented through RTX Lidar or Physics Lidar.

**Reply by yyjj25** [reply]
I have the same issue. May I ask how you resolved it?
I want to deploy the policy to Isaac Sim, but there is no RayCaster available to add in the observation.

**Reply by kellyguo11** [reply]
You can try using the RTX Lidar sensor to retrieve similar data as the RayCaster in Isaac Sim, or through the replicator point cloud annotator - https://docs.omniverse.nvidia.com/extensions/latest/ext_replicator/annotators_details.html#point-cloud

**Reply by ducmngx** [reply]
How can I get a 2D laser scan as part of my agent's observation?
